creator-node/test/recoverOrphanedData.jobProcessor.test.ts

```diff
@@ -0,0 +1,219 @@
+/* eslint-disable no-unused-expressions */
+import type { StateMonitoringUser } from '../src/services/stateMachineManager/stateMonitoring/types'
+
+import chai from 'chai'
+import sinon from 'sinon'
+import nock from 'nock'
+import proxyquire from 'proxyquire'
+import bunyan from 'bunyan'
+import _ from 'lodash'
+
+import recoverOrphanedDataJobProcessor from '../src/services/stateMachineManager/stateReconciliation/recoverOrphanedData.jobProcessor'
+import { QUEUE_NAMES } from '../src/services/stateMachineManager/stateMachineConstants'
+import { DecoratedJobReturnValue } from '../src/services/stateMachineManager/types'
+import { RecoverOrphanedDataJobReturnValue } from '../src/services/stateMachineManager/stateReconciliation/types'
+
+const { getApp } = require('./lib/app')
+const { getLibsMock } = require('./lib/libsMock')
+
+const models = require('../src/models')
+const config = require('../src/config')
+
+chai.use(require('sinon-chai'))
+const { expect } = chai
+
+describe('test recoverOrphanedData job processor', function () {
```

Comment:
could you modify test so that the queue has to execute multiple batches
```
const NUM_USERS_PER_QUERY = 10_000
const NUM_USERS_TO_RECOVER_PER_BATCH = 1000
```

wahtever easiest approach to doing this is

if not easy then dwbi but just calling out that as a gap

Reply:
fwiw I tested this locally and had a bug at first so I'm fairly confident in the pagination now. added an automated test for it now tho!

Reply:
dope